### DISCUSSION TITLE
bevy engine exports web files how to avoid being stolen in web pages

### DISCUSSION THREAD

**Discussion by ysypnbh** [bevyengine/bevy#16387] [Q&A]
I'm just worried that the things I finally made can be directly posted on other people's web pages.
I looked at the online tutorials
Found that wasm files and assets files are separated
If he obtains js and wasm files from the browser or crawler
He can turn it into another game by directly modifying the assets.
Is there any way to avoid
I don’t want to introduce too many dependencies, which will cause the exported file to be too large and slow to load.
My minimum requirement is that if the assets is changed, it cannot be run directly.
Is it possible to split wasm, just like a web framework, it can split multiple small files and load faster?

**Comment by tsal** [comment]
You should be able to do something like an md5sum on the assets content and store that as a const in your app; it's something that can be overcome  by bad actors with effort, but probably enough effort to discourage theft of a wasm-based game / program.

It can be a checksum of the entire contents, or a linch-pin file, such as the game logo. Something that, if you see it's changed, it's a good sign it's been stolen.
